third_party/move/move-vm/runtime/src/storage/ty_layout_converter.rs

```diff
@@ -157,6 +158,32 @@ pub(crate) trait LayoutConverterBase {
                 *count += 1;
                 self.struct_name_to_type_layout(*idx, ty_args, count, depth + 1)?
             },
+            Type::Function {
+                args,
+                results,
+                abilities,
+            } => {
+                let mut identifier_mapping = false;
+                let mut to_list = |rcs: &[triomphe::Arc<Type>]| {
+                    rcs.iter()
+                        .map(|rc| {
+                            self.type_to_type_layout_impl(rc.as_ref(), count, depth + 1)
```

Comment:
This is wrong and can cause OOM I believe: you need to call `*count += 1;` for this node itself, before you convert inner types.